pkg/githubclient/client_repository.go

```diff
@@ -201,12 +197,10 @@ func (c *Client) SetRepositoryBranchProtection(ctx context.Context, repository *
 	b, _ := json.MarshalIndent(opts, "", "  ")
 	c.logger.Debugf("branch protection settings\n%s", b)
 
-	if !c.dryRun {
-		underlyingClient := c.getUnderlyingClient(ctx)
-		_, _, err = underlyingClient.Repositories.UpdateBranchProtection(ctx, owner, repo, default_branch, opts)
-		if err != nil {
-			return microerror.Mask(err)
-		}
+	underlyingClient := c.getUnderlyingClient(ctx)
+	_, _, err = underlyingClient.Repositories.UpdateBranchProtection(ctx, owner, repo, default_branch, opts)
```

Comment:
So every repository touched gets branch protection enabled? Is this really a good idea (it feels quite opinionated)?
```suggestion
	_, _, err = underlyingClient.Repositories.UpdateBranchProtection(ctx, owner, repo, default_branch, opts)
```

Reply:
I did not modify the behavior, just removed the dry run

Reply:
OK, then I guess it's intended/approved.